src/Time/Rational.hs

```diff
@@ -170,3 +176,14 @@ class KnownRat (r :: Rat) where
 
 instance (KnownNat a, KnownNat b) => KnownRat (a :% b) where
     ratVal = natVal (Proxy @a) :% natVal (Proxy @b)
+
+
+newtype KnownRatDict (unit :: Rat) r = MkKnownRatDict (KnownRat unit => r)
+
+giftRat :: forall (unit :: Rat) r . (KnownRat unit => r) -> RatioNat -> r
+giftRat given = unsafeCoerce (MkKnownRatDict given :: KnownRatDict unit r)
+{-# INLINE giftRat #-}
+
+-- | Performs action with introduced 'DivRat' constraint for rational numbers.
+withDivRat :: forall (a :: Rat) (b :: Rat) r . (KnownRat a, KnownRat b) => (KnownRat (a / b) => r) -> r
+withDivRat r = giftRat @(a / b) r (ratVal @a / ratVal @b)
```

Comment:
Maybe also add `{-# INLINE #-}` pragmas to `withDivRat` and `toUnit` functions? I think it's a good idea in general to have `{-# INLINE #-}` pragmas for all our functions, like `threadDelay`, `getCPUTime`, `timeout`, etc.